### PR TITLE
Phoenix Controller Integration

### PR DESCRIPTION
**Pull request by bruce**
Experimental feature: Use GraphQL from Phoenix actions.

## Status

- [x] Basic functionality using string params
- [x] Test harness
- [x] Support for coercing non-string params (for integers, etc)
- [x] Support for input object params
- [x] Support for `@graphql` definitions using another Absinthe schema
- [x] Custom callback for pipeline building
- [x] Docs

Related features planned for future PRs:

- Helpers that sit on top of `Phoenix.HTML.Form` (validating adherence to input objects, etc)

## Example

Read the `@moduledoc` for `Absinthe.Phoenix.Controller`

### CODE REVIEW

**Comment by bruce** [comment]
@benwilson512  Should this wait to be merged until DocumentProvider makes it into Absinthe proper, so we don't have a dep on absinthe_plug?

**Comment by benwilson512** [comment]
Yup, that'll be in soon.

**Comment by bruce** [comment]
Okay, give this a look in the meantime. I'm sure there's opportunities for cleanup and clarification.

File: lib/absinthe/phoenix/controller.ex [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+  dialect. The following are the differences you need to keep in mind.
+
+  ### Objects can be leaf nodes
+
+  Let's look at the `users` example mentioned before:
+
+  ```
+  @graphql \"""
+    query ($filter: UserFilter) {
+      users(filter: $filter, limit: 10)
```
Comment (benwilson512) [inline lib/absinthe/phoenix/controller.ex:98]: @bruce how does this work with associations? IE what if I do 
```
{
  users(filter: $filter, limit: 10) {
    organization
  }
}
```

Does this only show the organization now? Or does it always return any non null value present?

Comment (bruce) [reply lib/absinthe/phoenix/controller.ex:98]: As we discussed earlier today: currently if you add a selection for a field (eg, as `users` has here), you don't get the full structs there. If we modify this to use explicit directives (or a similar mechanism), perhaps we can support both. Let's see after we get more feedback.

**Comment by bruce** [comment]
Merging this for inclusion in the next RC, hopefully to get more feedback and shake out any bugs before stable.

**Pull request merged**